nengo/builder/connection.py

```diff
@@ -107,6 +107,16 @@ def build_decoders(model, conn, rng):
     eval_points = get_eval_points(model, conn, rng)
     targets = get_targets(conn, eval_points)
 
+    if conn.solver.weights and not conn.solver.compositional:
+        # solver is solving for the whole weight matrix, so apply
+        # transform/encoders to targets
+        if not isinstance(conn.transform, Dense):
+            raise BuildError(
+                "Non-compositional solvers only work with Dense transforms")
+        targets = np.dot(targets, conn.transform.init.T)
```

Comment:
The transform could be a distribution. Furthermore, to ensure that the sampled transform ends up being the same whether we use a we use a weight solver or not, I think we should sample it in `build_connection` so that it gets sampled the same either way.

Reply:
Changed it so this is properly handled if transform is a Distribution.  I didn't move the transform sampling into `build_connection`, because I think we want the Transform build functions to be able to control that sampling (since they are the "authority" on the transform, and might need to customize that behaviour).  But actually the way `build_connection` works out, `rng` is only either passed to the Transform builder or this function, so the random state will be the same in both cases.